fhir-persistence-jdbc/src/test/java/com/ibm/fhir/persistence/jdbc/test/util/NewNumberParmBehaviorUtilTest.java

```diff
@@ -0,0 +1,467 @@
+        return parameterValue;
+    }
+
+    private QueryParameter generateParameter(SearchConstants.Prefix prefix, SearchConstants.Modifier modifier,
+            String code, String... values) {
+        QueryParameter parameter = new QueryParameter(SearchConstants.Type.NUMBER, code, modifier, null);
+        for (String value : values) {
+            parameter.getValues().add(generateParameterValue(value, prefix));
+        }
+        return parameter;
+    }
+
+    private QueryParameter generateParameter(SearchConstants.Prefix prefix, SearchConstants.Modifier modifier,
+            String value) {
+        return generateParameter(prefix, modifier, "precision", new String[] { value });
+    }
+
+    private QueryParameter generateParameter(SearchConstants.Prefix prefix, SearchConstants.Modifier modifier,
+            String... values) {
+        return generateParameter(prefix, modifier, "precision", values);
+    }
+
+    private void runTest(QueryParameter queryParm, List<Object> expectedBindVariables, String expectedSql)
+            throws FHIRPersistenceException {
+        runTest(queryParm, expectedBindVariables, expectedSql, "Basic", Basic.class);
+    }
+
+    private void runLowerUpperTest(String value, String lower, String upper) {
+        BigDecimal decimal = new BigDecimal(value);
+        String actualLower = NewNumberParmBehaviorUtil.generateLowerBound(decimal).toPlainString();
+        assertEquals(actualLower, lower);
+
+        String actualUpper = NewNumberParmBehaviorUtil.generateUpperBound(decimal).toPlainString();
+        assertEquals(actualUpper, upper);
+    }
+
+    private void runTest(QueryParameter queryParm, List<Object> expectedBindVariables, String expectedSql,
+            String tableAlias, Class<?> resourceType)
+            throws FHIRPersistenceException {
+        WhereFragment actualWhereClauseSegment = new WhereFragment();
+
+        NewNumberParmBehaviorUtil.executeBehavior(actualWhereClauseSegment, queryParm, tableAlias);
+        List<BindMarkerNode> collectBindMarkersInto = new ArrayList<>();
+        StringExpNodeVisitor visitor = new StringExpNodeVisitor(null, collectBindMarkersInto, false);
+        final String actualWhereClauseString = actualWhereClauseSegment.getExpression().visit(visitor);
+
+        if (log.isLoggable(LOG_LEVEL)) {
+            log.info("whereClauseSegment -> " + actualWhereClauseString);
+            log.info("bind variables -> " + collectBindMarkersInto);
+        }
+        assertEquals(actualWhereClauseString, expectedSql);
+        assertEquals(collectBindMarkersInto.size(), expectedBindVariables.size());
+
+        // TODO check bind values are correct
+//        for (Object o : expectedBindVariables) {
+//            actualBindVariables.remove(o);
+//        }
+//
+//        if (log.isLoggable(LOG_LEVEL)) {
+//            log.info("leftover - bind variables -> " + actualBindVariables);
+//        }
+//        assertEquals(actualBindVariables.size(), 0);
+
```

Comment:
? what the comment for?

Reply:
Implemented a proper check, and found that some of the tests were giving the bind variables in the wrong order (the previous implementation didn't care about the order).